core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -264,11 +268,12 @@ object ConsumerPerformance {
           val messageAndMetadata = iter.next
           messagesRead += 1
           bytesRead += messageAndMetadata.message.length
+          val currentTimeMillis = System.currentTimeMillis
 
-          if (messagesRead % config.reportingInterval == 0) {
+          if (currentTimeMillis - lastReportTime >= config.reportingInterval) {
```

Comment:
For reporting purpose, I think it's enough to call System.currentTimeMillis once per while loop, instead of once per record. Also, the patch changed the behavior of the tool a bit. Currently, if there are no new messages, the tool will exit after 1 sec. With the patch, the tool will block forever. We will need to revert to the previous behavior.

Finally, since our system tests use this tool. Could you run our system tests (see test/README.md) and make sure that the patch doesn't break any test? If needed, @granders can give you access to our jenkins job so that you can test your branch on EC2.